pkg/library/libraryapplyconfiguration/operator_launch_helpers.go

```diff
@@ -61,19 +64,15 @@ func (a SimpleOperatorStarter) RunOnce(ctx context.Context) error {
 		return fmt.Errorf("the following controllers were requested to run multiple times: %v", duplicateControllerNames)
 	}
 
-	controllersToRunSet := sets.NewString(a.Controllers...)
-	if controllersToRunSet.Len() == 0 {
-		controllersToRunSet = knownControllersSet
-	}
-	if unknownControllersToRun := controllersToRunSet.Difference(knownControllersSet); len(unknownControllersToRun) > 0 {
-		return fmt.Errorf("requested unknown controllers to be run: %v, known controllers: %v", unknownControllersToRun.List(), knownControllersSet.List())
+	if errs := validateControllersFromFlags(knownControllersSet, a.Controllers); len(errs) > 0 {
```

Comment:
validation happens late because the list of controllers to run is not known beforehand.